src/modules/uORB/uORBDeviceNode.cpp

```diff
@@ -661,12 +635,23 @@ int uORB::DeviceNode::update_queue_size(unsigned int queue_size)
 		return PX4_OK;
 	}
 
-	//queue size is limited to 255 for the single reason that we use uint8 to store it
-	if (_data || _queue_size > queue_size || queue_size > 255) {
+	if (_queue_size > queue_size || queue_size > 255) {
+		// queue size is limited to 255 for the single reason that we use uint8 to store it
+		return PX4_ERROR;
+	}
+
+	uint8_t *data = new uint8_t[_meta->o_size * queue_size];
+
+	if (data == nullptr) {
 		return PX4_ERROR;
 	}
 
+	// switch to new data
+	memcpy(data, _data, _meta->o_size * _queue_size);
```

Comment:
Increasing the queue size does not work: due to the `generation % _queue_size` logic, a subscriber could access invalid data after this update.
Better keep the existing logic until we move the queue size to the meta-data.